includes/class-admin-settings.php

```diff
@@ -0,0 +1,516 @@
+	 */
+	public function get_setting( $setting_name, $default_value = false ) {
+		if ( null === $this->options ) {
+			$options = get_option( $this->option_name, false );
+			/**
+			 * If the options are not set load defaults.
+			 */
+			if ( false === $options ) {
+				$options = $this->get_default_settings();
+				update_option( $this->option_name, $options );
+			}
+			$config_file_options = $this->get_settings_from_config_file();
+			if ( is_array( $options ) ) {
+				/**
+				 * If User Options are saved do some processing to make it match the structure of the data from the config file.
+				 */
+				if ( isset( $options['enable_debug_type'] ) && is_array( $options['enable_debug_type'] ) ) {
+					$debug_types = array();
+					foreach ( $options['enable_debug_type'] as $debug_type_name => $debug_type_enabled ) {
+						if ( $debug_type_enabled ) {
+							$debug_types[] = $debug_type_name;
+						}
+					}
+					$options['enable_debug_type'] = $debug_types;
+				}
+				$this->options = wp_parse_args( $config_file_options, $options );
+			}
+		}
+		return ( isset( $this->options[ $setting_name ] ) ? $this->options[ $setting_name ] : $default_value );
+	}
+
+	/**
+	 * Get the values defined in the config file.
+	 *
+	 * @return array An array of values as defined in the Config File.
+	 */
+	private function get_settings_from_config_file() {
+		$options = array();
+
+		if ( ! defined( 'AP_ENABLE' ) ) {
+			define( 'AP_ENABLE', false );
+		} elseif ( AP_ENABLE ) {
+			$options['enable'] = AP_ENABLE;
+		}
+
+		if ( ! defined( 'AP_API_KEY' ) ) {
+			define( 'AP_API_KEY', '' );
+		} else {
+			$options['api_key'] = AP_API_KEY;
+		}
+
+		if ( ! defined( 'AP_HOSTS' ) ) {
+			define( 'AP_HOSTS', array() );
+		} elseif ( is_array( AP_HOSTS ) ) {
+			$options['api_hosts'] = AP_HOSTS;
+		}
+
+		if ( ! defined( 'AP_DEBUG' ) ) {
+			define( 'AP_DEBUG', false );
+		} elseif ( AP_DEBUG ) {
+			$options['enable_debug'] = AP_DEBUG;
+		}
+
+		if ( ! defined( 'AP_DEBUG_TYPES' ) ) {
+			define( 'AP_DEBUG_TYPES', array() );
+		} elseif ( is_array( AP_DEBUG_TYPES ) ) {
+			$options['enable_debug_type'] = AP_DEBUG_TYPES;
+		}
+
+		if ( ! defined( 'AP_DISABLE_SSL' ) ) {
+			define( 'AP_DISABLE_SSL', false );
+		} elseif ( AP_DISABLE_SSL ) {
+			$options['disable_ssl_verification'] = AP_DISABLE_SSL;
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Register the Admin Menu.
+	 *
+	 * @return void
+	 */
+	public function register_admin_menu() {
+		add_options_page(
+			'AspirePress',
+			'AspirePress',
+			'manage_options',
+			'aspirepress-settings',
+			array( $this, 'the_settings_page' )
+		);
+	}
+
+	/**
+	 * Enqueue the Scripts and Styles.
+	 *
+	 * @param string $hook The page identifier.
+	 * @return void
+	 */
+	public function admin_enqueue_scripts( $hook ) {
+		if ( 'settings_page_aspirepress-settings' !== $hook ) {
+			return;
+		}
+		wp_enqueue_style( 'aspire_update_settings_css', plugin_dir_url( __DIR__ ) . 'assets/css/aspire-update.css', array(), '1.0' );
```

Comment:
Ideally version here (and in other scripts) at some point becomes a dynamic $version so not every part of the code needs edit but just one $version variable or however it's decided to be handled (global, container service, etc). But this can be done in future too.

Reply:
I usually define a constant and use?  Agree to do it that way?

Reply:
I dislike constants but the only alternative are container services or per-file $vars.
The first is too much work for now, the second is dirt. So I guess Constant is fine as long its in our namespace (prefixed)

Going forward I recommend a container service with singleton config class that can be used allover the code.

Reply:
Added Constant to define version.